src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/ImageService.java

```diff
@@ -0,0 +1,167 @@
+package de.terrestris.shogun2.service;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+
+import javax.imageio.ImageIO;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.imgscalr.Scalr;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import de.terrestris.shogun2.dao.ImageDao;
+import de.terrestris.shogun2.model.Image;
+
+/**
+ * Service class for the {@link MapMavinUserProfilePicture} model.
+ *
+ * @author Daniel Koch
+ * @author Johannes Weskamm
+ *
+ */
+@Service("imageService")
+public class ImageService<E extends Image, D extends ImageDao<E>>
+		extends FileService<E, D> {
+
+	/**
+	 * We have to use {@link Qualifier} to define the correct dao here.
+	 * Otherwise, spring can not decide which dao has to be autowired here
+	 * as there are multiple candidates.
+	 */
+	@Override
+	@Autowired
+	@Qualifier("imageDao")
+	public void setDao(D dao) {
+		this.dao = dao;
+	}
+
+	/**
+	 * Method persists a given Image as a bytearray in the database
+	 *
+	 * @param file
+	 * @param resize
+	 * @param imageSize
+	 * @return
+	 * @throws Exception
+	 */
+	@SuppressWarnings("unchecked")
+	public Image uploadImage(MultipartFile file, boolean resize, Integer dimensions)
+			throws Exception {
```

Comment:
Arugment `resize` is misleading, maybe we can rename it to `createThumbnail` or something else?